Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+    {
+        AZ::Interface<IMultiplayerConnectionViewportMessage>::Register(this);
+    }
+
+    MultiplayerConnectionViewportMessageSystemComponent::~MultiplayerConnectionViewportMessageSystemComponent()
+    {
+        AZ::Interface<IMultiplayerConnectionViewportMessage>::Unregister(this);
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("MultiplayerService"));
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void MultiplayerConnectionViewportMessageSystemComponent::OnTick(float, AZ::ScriptTimePoint)
+    {
+        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!viewport)
+        {
+            return;
+        }
+
+        // Display the custom center viewport text
+        if (!m_centerViewportDebugText.empty())
+        {            
+            AzFramework::WindowSize viewportSize = viewport->GetViewportSize();
+            const float dpiScalingFactor = viewport->GetDpiScalingFactor();
+            const float center_screenposition_x = 0.5f*viewportSize.m_width;
+            const float center_screenposition_y = 0.5f*viewportSize.m_height;
+            const float screenposition_title_y = center_screenposition_y-9;
+            const float screenposition_debugtext_y = screenposition_title_y+18*dpiScalingFactor;
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::Yellow);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_title_y / dpiScalingFactor, editorsv_connectionMessageFontSize, "Multiplayer Editor", true);
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::White);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_debugtext_y / dpiScalingFactor, editorsv_connectionMessageFontSize, m_centerViewportDebugText.c_str(), true);
+        }
+
+        // Build the connection status string (just show client connected or disconnected status for now)
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();        
+        if (agentType != MultiplayerAgentType::Client)
```

Comment:
Curious: Do we want to display the text for ClientServer agent type?

Reply:
Nothing for clientserver or dedicated server. Need to think about what that would look like... for example, showing connection status for each client? Not sure.